.github/workflows/pytest.yml

```diff
@@ -198,11 +198,30 @@ jobs:
           path: coverage
       - name: List downloaded files
         run: |
-          ls -R
+          find coverage -type f
       - name: Upload test coverage report to CodeCov
         uses: codecov/codecov-action@v3
         with:
           directory: coverage
+      - name: Install Micromamba
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          init-shell: bash
+          environment-name: coverage
+          create-args: >-
+            python=3.11
+            coverage>=7.4.1
+      - name: Combine all coverage data into a single coverage.xml
+        run: |
+          micromamba run -n coverage coverage combine coverage/*/.coverage
+          micromamba run -n coverage coverage xml
+      - name: Check coverage
+        uses: orgoro/coverage@v3.1
+        with:
+          coverageFile: coverage.xml
+          token: ${{ secrets.GITHUB_TOKEN }}
+          # Set impossibly high to test failure condition in merge queue
+          thresholdAll: 1.0
```

Comment:
Install coverage, use it to combine the `.coverage` artifacts and convert them into a unified `coverage.xml` and then use the `orgoro/coverage` action to check whether we met the threshold value.